### ISSUE TITLE
Merge 'with' updates

### ISSUE THREAD

**Issue by Timmmm** [riscv/sail-riscv#583]
There's quite a lot of code like this:

```
function lower_mstatus(m : Mstatus) -> Sstatus = {
  let s = Mk_Sstatus(zero_extend(0b0));
  let s = [s with SD = m[SD]];
  let s = set_sstatus_UXL(s, get_mstatus_UXL(m));
  let s = [s with MXR = m[MXR]];
  let s = [s with SUM = m[SUM]];
  let s = [s with XS = m[XS]];
  let s = [s with FS = m[FS]];
  let s = [s with VS = m[VS]];
  let s = [s with SPP = m[SPP]];
  let s = [s with SPIE = m[SPIE]];
  let s = [s with UPIE = m[UPIE]];
  let s = [s with SIE = m[SIE]];
  let s = [s with UIE = m[UIE]];
  s
}
```

which should be written like this:

```
function lower_mstatus(m : Mstatus) -> Sstatus = {
  let s = Mk_Sstatus(zero_extend(0b0));
  [s with
    SD = m[SD],
...
    SPP = m[SPP],
    SPIE = m[SPIE],
    UPIE = m[UPIE],
    SIE = m[SIE],
    UIE = m[UIE],
  ]
}
```

**Comment by jordancarlin** [comment]
Can be closed with #586 merged

**Comment by Timmmm** [comment]
I think there's a few more that could be merged:

* `legalize_mstatus`
* `lower_mstatus` (mostly done but we missed `SD`)
* `lift_sie`
